src/client.js

```diff
@@ -243,11 +243,15 @@ class Client {
     let dnsFailed = false
     if (typeof key === 'string') {
       cabalPromise = this.resolveName(key.trim()).then((resolvedKey) => {
-        // discard uri scheme and search params of cabal key, if present. returns 64 chr hex string
+        if (resolvedKey === null) {
+          dnsFailed = true
+          return
+        }
+
+        // verify that scrubbedKey is 64 ch hex string
         const scrubbedKey = Client.scrubKey(resolvedKey)
 
-        //  verify that scrubbedKey is 64 ch hex string
-        if (resolvedKey === null || !Cabal.isHypercoreKey(scrubbedKey)) {
```

Comment:
This `null` check should be done before we try to scrub the key to avoid `scrubKey()` from blowing up.